src/order/bounded_order.lean

```diff
@@ -1813,6 +1812,66 @@ lemma inf_sup {x' y'} (h : is_compl x y) (h' : is_compl x' y') :
 
 end is_compl
 
+namespace prod
+
+lemma disjoint_iff [partial_order α] [partial_order β] [order_bot α] [order_bot β] {x y : α × β} :
+  disjoint x y ↔ disjoint x.1 y.1 ∧ disjoint x.2 y.2 :=
+begin
+  split,
+  { intros h,
+    refine ⟨λ a hx hy, (@h (a, ⊥) ⟨hx, _⟩ ⟨hy, _⟩).1, λ b hx hy, (@h (⊥, b) ⟨_, hx⟩ ⟨_, hy⟩).2⟩,
+    all_goals { exact bot_le }, },
+  { rintros ⟨ha, hb⟩ z hza hzb,
+    refine ⟨ha hza.1 hzb.1, hb hza.2 hzb.2⟩ },
+end
+
+lemma codisjoint_iff [partial_order α] [partial_order β] [order_top α] [order_top β] {x y : α × β} :
+  codisjoint x y ↔ codisjoint x.1 y.1 ∧ codisjoint x.2 y.2 :=
+@disjoint_iff αᵒᵈ βᵒᵈ _ _ _ _ _ _
+
+lemma is_compl_iff [partial_order α] [partial_order β] [bounded_order α] [bounded_order β]
+  {x y : α × β} :
+  is_compl x y ↔ is_compl x.1 y.1 ∧ is_compl x.2 y.2 :=
+by simp_rw [is_compl_iff, disjoint_iff, codisjoint_iff, and_and_and_comm]
+
+end prod
+
+namespace pi
+variables {ι : Type*} {α' : ι → Type*}
+
+lemma disjoint_iff [Π i, partial_order (α' i)] [Π i, order_bot (α' i)] {f g : Π i, α' i} :
+  disjoint f g ↔ ∀ i, disjoint (f i) (g i) :=
+begin
+  split,
+  { intros h i x hf hg,
+    refine (update_le_iff.mp $
+      h (update_le_iff.mpr ⟨hf, λ _ _, _⟩) (update_le_iff.mpr ⟨hg, λ _ _, _⟩)).1,
+    { exact ⊥},
+    { exact bot_le },
+    { exact bot_le }, },
+  { intros h x hf hg i,
+    apply h i (hf i) (hg i) },
+end
+
+lemma codisjoint_iff [Π i, partial_order (α' i)] [Π i, order_top (α' i)] {f g : Π i, α' i} :
+  codisjoint f g ↔ ∀ i, codisjoint (f i) (g i) :=
+@disjoint_iff _ (λ i, (α' i)ᵒᵈ) _ _ _ _
+
+lemma is_compl_iff [Π i, partial_order (α' i)] [Π i, bounded_order (α' i)] {f g : Π i, α' i} :
```

Comment:
```suggestion
protected lemma is_compl_iff [Π i, partial_order (α' i)] [Π i, bounded_order (α' i)] {f g : Π i, α' i} :
```
and the other two

Reply:
Are you suggesting doing this for all 9 lemmas, just the `pi` ones, or just the `is_compl_iff` ones? I'm not sure I see much point, we don't typically `open prod`, `open pi`, or `open Prop`.

Reply:
`open prod` is done twice at least, actually. I'm not sure about the other two.

Reply:
So which lemmas do you want `protected` on?

Reply:
The `prod` ones

Reply:
Despite the fact that you put this PR comment on the `pi` ones?

Reply:
Uh, that was accidental.